### PR TITLE
Added delay_pre function and bnw function

### PR DESCRIPTION
**Pull request by aditmehta9**
Hi Sir,
I am Adit Mehta, GSoC applicant, interested in contributing RMagick project, first-year student of DAIICT. While I was learning RMagick I was amazed by the ability to create GIF. I got to know that delay feature was adding same time to every picture of GIF but then I thought of adding a new function which will able to give time to every picture specifically according to the user.

### CODE REVIEW

**Comment by vassilevsky** [comment]
Thanks, Adit. Can you also write tests for this function? You can use RSpec if you want. See CONTRIBUTING.md for instructions.

**Comment by aditmehta9** [comment]
Yes, sir I am new to this I am not that much familiar with RSpec please provide me guidance.

**Comment by aditmehta9** [comment]
Sir @mockdeep please help me, how I could create tests using RSpec? I am new to this.

**Comment by aditmehta9** [comment]
@vassilevsky Thanks, sir

**Comment by aditmehta9** [comment]
Sir, I have added RSpec tests.

**Comment by vassilevsky** [comment]
Thanks, Adit. I want to see how well your tests cover your functions. For that I have to set up coverage tracking. This repository already sends data to CodeClimate, but not coverage yet. I will add coverage there and then it will be easier to review all pull requests. I will try to do this ASAP.

**Comment by aditmehta9** [comment]
Okay, Sir

**Comment by aditmehta9** [comment]
Sir, @vassilevsky are there any changes I need to do in pull request??

**Comment by aditmehta9** [comment]
Sir, are there any issues in commit that i should fix?

**Comment by aditmehta9** [comment]
Sir, are there any changes I need to make?

**Comment by aditmehta9** [comment]
Sir, are there any changes that I need to make?

File: lib/rmagick_internal.rb [hunk 1]
```diff
@@ -1535,6 +1535,16 @@ def delay=(d)
       end
       @images.each { |f| f.delay = Integer(d) }
     end
+    
+    # Function which will able to give time to every picture specifically according to the user.
+    def delay_pre=(d)
+       # Time Unit: milliseconds
+       if Integer(d) < 0
+         fail ArgumentError, 'delay must be greater than or equal to 0'
+       end
```
Comment (vassilevsky) [inline lib/rmagick_internal.rb]: Please add a test for this case.

File: lib/rmagick_internal.rb [hunk 2]
```diff
@@ -1586,6 +1596,20 @@ def fill(*args, &block)
...
+      @pixels = images.get_pixels(0, 0, images.columns, images.rows)
+         for @pixel in @pixels
+                  avg = (@pixel.red + @pixel.green + @pixel.blue) / 3
+                  @pixel.red = avg
+                  @pixel.green = avg
+                  @pixel.blue = avg
+         end
+      images.store_pixels(0, 0, images.columns, images.rows, @pixels)
+           return images
+    end
```
Comment (vassilevsky) [inline lib/rmagick_internal.rb]: Please add a test of this method.

**Comment by vassilevsky** [comment]
Github says that this branch has conflicts that must be resolved. So, I can't merge your branch to this project, even if I want. Could you resolve them?

Here is a description of the process:

https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/

**Comment by mockdeep** [comment]
Hi @aditmehta9, thanks so much for the time you spent on this! Unfortunately, I don't think it's a great fit for RMagick. The `delay_pre=` method doesn't seem to map to anything in ImageMagick and it looks like something that is for a very specific use case. As for black and white, you can already accomplish this using RMagick:

```rb
image.quantize(256, Magick::GRAYColorspace)
```

This is likely to be much faster, as well, since it is written in C instead of Ruby. Sorry you had to spend so much time on this only to have it rejected. Hopefully it doesn't discourage you from contributing to open source in the future.

**Comment by aditmehta9** [comment]
Thank you, sir, for your feedback.

**Pull request closed without merging**